pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/withscript/WithScriptDescriptor.java

```diff
@@ -39,6 +46,10 @@
  */
 public abstract class WithScriptDescriptor<T extends WithScriptDescribable<T>> extends Descriptor<T> {
 
+    protected WithScriptDescriptor() {
+        ExtensionList.lookupSingleton(WithScriptAllowlist.class).registerScript(getScriptResource());
```

Comment:
SpotBugs might complain that you are calling an overridable method from a constructor?

Reply:
Yeah, I was surprised that it does not complain about this. Maybe it is because `getScriptResource` is `protected`, so SpotBugs assumes the behavior is intentional.